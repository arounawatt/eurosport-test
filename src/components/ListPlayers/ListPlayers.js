import React, { useEffect, useState } from 'react';
import _get from "lodash/get";
import axios from 'axios';

import "./ListPlayers.scss";
import PlayerCard from '../PlayerCard/PlayerCard';

const ListPlayers = () => {

  const [players, setPlayers] = useState(null);
  const [error, setError] = useState(null);

  const getPlayers = () => {
    axios.get("https://eurosportdigital.github.io/eurosport-web-developer-recruitment/headtohead.json").then(
      ({data}) => {
        setPlayers(_get(data, "players", []));
      },
      () => {
        setError(true);
      }
    );
  };

  useEffect(() => {
    getPlayers();
  }, []);

  return (
    <div className="ListPlayers">
      {players && players.map( player => (
        <PlayerCard key={_get(player, "data.rank")} item={player} />
      ))}
      {error && (
        <span> A error occur. please check your connection or retry later</span>
      )}
    </div>
  );
};

export default ListPlayers;