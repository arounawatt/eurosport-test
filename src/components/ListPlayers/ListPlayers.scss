.ListPlayers {
	display: flex;
	flex-direction: column;
	margin: $space-s;

	@include gridle-state(medium) {
		flex-direction: row;
	}
}