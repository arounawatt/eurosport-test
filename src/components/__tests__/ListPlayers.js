import React from "react"
import ReactDOM from 'react-dom';
import axios from "axios"
import { act } from 'react-dom/test-utils';
import playersMock from "../../../__mocks__/players";
import ListPlayers from "../ListPlayers/ListPlayers";

jest.mock('axios');
jest.mock('../PlayerCard/PlayerCard', () => "mock-player-card");

let container;

beforeEach(() => {
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  document.body.removeChild(container);
  container = null;
});

describe("ListPLayers", () => {
  it("should render correctly", async () => {
    axios.get = jest.fn().mockReturnValue(Promise.resolve(playersMock));
     act(() => {
      ReactDOM.render(<ListPlayers />, container);
    });
    await Promise.resolve();
    expect(container.innerHTML).toMatchSnapshot()
  })

  it("should render error", async () => {
    axios.get = jest.fn().mockReturnValue(Promise.reject());
     act(() => {
      ReactDOM.render(<ListPlayers />, container);
    });
    await Promise.resolve();
    expect(container.innerHTML).toMatchSnapshot()
  })
})