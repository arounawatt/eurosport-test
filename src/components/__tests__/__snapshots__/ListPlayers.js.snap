// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListPLayers should render correctly 1`] = `"<div class=\\"ListPlayers\\"><mock-player-card item=\\"[object Object]\\"></mock-player-card><mock-player-card item=\\"[object Object]\\"></mock-player-card></div>"`;

exports[`ListPLayers should render error 1`] = `"<div class=\\"ListPlayers\\"><span> A error occur. please check your connection or retry later</span></div>"`;
